src/repr/src/adt/timestamp.rs

```diff
@@ -494,6 +506,106 @@ impl<T: TimestampLike> CheckedTimestamp<T> {
     pub fn checked_sub_signed(self, rhs: Duration) -> Option<T> {
         self.t.checked_sub_signed(rhs)
     }
+
+    /// Implementation ported from Postgres's `timestamp.c`.
+    ///
+    /// <https://github.com/postgres/postgres/blob/517bf2d91/src/backend/utils/adt/timestamp.c#L3565>
```

Comment:
Since this is a straightforward port, we may need to split this out to a different file with different licensing?

Reply:
Good question! I'm not really sure what we're supposed to do here. If we need to move this function to a different file and add the PostgreSQL license, I'm more than happy to :)

Reply:
I'm not really sure either... @benesch, any suggestion?

Reply:
No need to split out a different file! Just update the copyright header like this:

https://github.com/MaterializeInc/materialize/blob/ee747b414cdb3f726de72a7a4fa8bc1feb6881ba/src/ore/src/netio/read_exact.rs#L17-L23

Reply:
Thanks for the info! Updated!